src/schema/v2/sale/index.ts

```diff
@@ -278,6 +278,35 @@ export const SaleType = new GraphQLObjectType<any, ResolverContext>({
         resolve: ({ require_identity_verification }) =>
           require_identity_verification,
       },
+      userNeedsIdentityVerification: {
+        type: GraphQLBoolean,
+        description:
+          "True if the current user needs to undergo identity verification for this sale, false otherwise",
+        resolve: (
+          { require_identity_verification, id },
+          _options,
+          { meLoader, meBiddersLoader }
+        ) => {
+          if (require_identity_verification) {
+            if (!meLoader || !meBiddersLoader) {
+              return true
+            } else {
+              return meBiddersLoader({ sale_id: id }).then(bidders => {
+                if (bidders.length > 0) {
+                  const bidder = bidders[0]
+                  return bidder.needs_identity_verification
+                } else {
+                  return meLoader().then(({ identity_verified }) => {
+                    return !identity_verified
+                  })
+                }
+              })
+            }
+          } else {
+            return false
+          }
+        },
+      },
```

Comment:
I would flatten out all the conditionals to make this more legible

```tsx
if (!require_identity_verification) {
  return false;
}

if (!meLoader || !meBiddersLoader) {
  return true;
}

return meBiddersLoader({ sale_id: id }).then((bidders) => {
  if (bidders.length > 0) {
    const [bidder] = bidders;
    return bidder.needs_identity_verification;
  }

  return meLoader().then(({ identity_verified }) => {
    return !identity_verified;
  });
});

```

----

Though I can't really speak to any of the other logic.

Reply:
+1 - thanks for the input here, will refactor accordingly.

Reply:
Addressed in https://github.com/artsy/metaphysics/pull/2332/commits/5406a988de80fc9d5544360718b576157a0bc580 – personally, I don't like to go full on guard class, so left a single `if / else` expression in.